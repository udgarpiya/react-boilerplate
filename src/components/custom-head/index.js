import React from 'react'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'

const CustomHead = ({
  title = 'React Boilerplace',
  description = 'React Boilerplate Description',
}) => {
  return (
    <Helmet>
      <title>{title}</title>
      <meta name="description" content={description} />
    </Helmet>
  )
}

CustomHead.defaultProps = {
  title: 'React Boilerplace',
  description: 'React Boilerplate Description',
}

CustomHead.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
}

export { CustomHead }
export default CustomHead
