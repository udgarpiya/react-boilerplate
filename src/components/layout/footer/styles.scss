.footer-component {
  &__main-container {
    padding: 0 15px;
    height: 50px;
    justify-content: center;
    display: flex;
    align-items: center;
    z-index: 1;
  }
}