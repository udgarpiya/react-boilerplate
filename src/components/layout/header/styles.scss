.header-component {
  &__main-container {
    padding: 0 15px;
    height: 50px;
    justify-content: center;
    display: flex;
    align-items: center;
    position: sticky;
    z-index: 1;
    top: -1px;
    left: 0;
    box-shadow: 0px 0px 5px 1px rgba(0, 0, 0, 0.15);
  }
}