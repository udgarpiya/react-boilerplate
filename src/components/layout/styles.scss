.layout-component {
  &__children-container {
    min-height: calc(100vh - 100px);
  }
}