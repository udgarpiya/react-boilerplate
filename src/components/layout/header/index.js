import React from 'react'
import './styles.scss'

const Header = () => {
  return (
    <div className="header-component__main-container">
      <h1>Header Component</h1>
    </div>
  )
}

export { Header }
export default Header
