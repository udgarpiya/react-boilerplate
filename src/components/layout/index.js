import React from 'react'
import PropTypes from 'prop-types'

import Header from './header'
import Footer from './footer'
import CustomHead from '../custom-head'
import './styles.scss'

const Layout = ({ children, title, description }) => {
  return (
    <div className="layout-component__main-component">
      <CustomHead title={title} description={description} />
      <Header />
      <div className="layout-component__children-container">{children}</div>
      <Footer />
    </div>
  )
}

Layout.defaultProps = {
  title: '',
  description: '',
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
}

export { Layout }
export default Layout
