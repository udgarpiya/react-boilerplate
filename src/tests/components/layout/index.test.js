import React from 'react'
import { shallow } from 'enzyme'
import { Layout } from 'components/layout'
import { Header } from 'components/layout/header'
import { Footer } from 'components/layout/footer'

describe('Layout Component', () => {
  let component

  const props = {
    children: <h1>Child component</h1>,
    title: 'title',
    description: 'description',
  }

  beforeEach(() => {
    component = shallow(<Layout {...props} />)
  })

  it('It should have main component and child component', () => {
    const LayoutMain = component.find('.layout-component__main-component')
    const ChildContainer = component.find(
      '.layout-component__children-container',
    )

    expect(LayoutMain.length).toBe(1)
    expect(ChildContainer.length).toBe(1)
    expect(ChildContainer.text()).toBe('Child component')
    expect(component).toMatchSnapshot()
  })

  it('It should have Header and Footer component', () => {
    expect(component.containsMatchingElement(<Header />)).toEqual(true)
    expect(component.children().find(Header).length).toEqual(1)
    expect(component.containsMatchingElement(<Footer />)).toEqual(true)
    expect(component.children().find(Footer).length).toEqual(1)
  })
})
