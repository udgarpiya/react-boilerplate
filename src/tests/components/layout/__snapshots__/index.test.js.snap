// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component It should have main component and child component 1`] = `
<div
  className="layout-component__main-component"
>
  <CustomHead
    description="description"
    title="title"
  />
  <Header />
  <div
    className="layout-component__children-container"
  >
    <h1>
      Child component
    </h1>
  </div>
  <Footer />
</div>
`;
